src/render/traverse.js

```diff
@@ -237,7 +237,9 @@ function traverse ({ seq, node, context, numChildren }) {
     return;
   }
 
-  if (node === null) {
+  // in JSX, all are valid children:
+  // null, undefined, false and true
+  if (node === true || node === false || node === undefined || node === null) {
```

Comment:
Some of these cases are already handled in `shouldEmitByType` which is called immediately above here. Not sure if correct behavior is to short-circuit there for `true` also, or to emit empty as is done here. See what React 's renderer does?